packages/jamtools/features/modules/dashboards/keytar_and_foot_dashboard/multi_octave_supervisor.tsx

```diff
@@ -0,0 +1,161 @@
+import React from 'react';
+
+import {ModuleAPI} from '~/core/engine/module_api';
+import {MidiEvent, MidiEventFull} from '~/core/modules/macro_module/macro_module_types';
+import {Button} from '~/core/components/Button';
+
+type MultiOctaveSupervisorMidiState = {
+    currentlyHeldDownInputNotes: MidiEvent[];
+};
+
+export class MultiOctaveSupervisor {
+    private macros!: Awaited<ReturnType<MultiOctaveSupervisor['createMacros']>>;
+    private states!: Awaited<ReturnType<MultiOctaveSupervisor['createStates']>>;
+    private actions!: ReturnType<MultiOctaveSupervisor['createActions']>;
+
+    private midiState: MultiOctaveSupervisorMidiState = {
+        currentlyHeldDownInputNotes: [],
+    };
+
+    private handleKeyboardNote = async (fullEvent: MidiEventFull) => {
+        const event = fullEvent.event;
+
+        this.macros.midiOutput.send(event);
```

Comment:
On this line we're sending the appropriate crafted midi event to the user's selected midi output(s). This code is agnostic to whether the user multiple outputs set up to receive these notes. It will automatically send it to all of them if there are multiple.